tests/models/modules/test_SwaVFreezing.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+
+import torch
+
+from lightly.models.modules.heads import SwaVPrototypes
+
+class TestSwavFreezingPrototypes(unittest.TestCase):
+    def test_swav_frozen_prototypes(self, device: str = "cpu", seed=0):
+        criterion = torch.nn.L1Loss()
+        linear_layer = torch.nn.Linear(8, 8, bias=False)
+        prototypes = SwaVPrototypes(8, 8, 2)
+        optimizer = torch.optim.SGD(prototypes.parameters(), lr=0.01)
+        for step in range(4):
+            torch.manual_seed(seed)
+            in_features = torch.rand(4, 8, device="cpu")
+            out_features = linear_layer(in_features)
+            target_features = torch.ones(4, 8, device="cpu")
```

Comment:
I think the features only have to be defined once. Same for setting the seed.

```suggestion
        torch.manual_seed(seed)
        in_features = torch.rand(4, 8, device="cpu")
        target_features = torch.ones(4, 8, device="cpu")
        for step in range(4):
            out_features = linear_layer(in_features)
```